configs/envoy_double_proxy.template.json

```diff
@@ -5,8 +5,8 @@
     "ssl_context": {
       "alpn_protocols": "h2,http/1.1",
       "alt_alpn_protocols": "http/1.1",
-      "cert_chain_file": "/etc/envoy/cert.pem",
-      "private_key_file": "/etc/envoy/key.pem"
+      "cert_chain_file": "certs/servercert.pem",
```

Comment:
Quick comment which unfortunately is going to make this more complicated: We are going to need to fix config_test to work with some type of mock loader that doesn't read files from disk. Optimally it would verify everything else in SSL and possibly just use fake valid cert data. The reason this is going to be needed is we run at Lyft config tests for configs where the certs are not available on the box where the tests are run.

Ultimately I think we know this is working when none of the example config stuff here changed as part of this diff.

Reply:
I have mixed feelings about validity of this approach, but I'll add this if you need it.

Reply:
I'm sorry, but we definitely need it. I would just figure out whatever is easiest in that test, even if you have to mock out more stuff.

Reply:
Sorry, I think I might have misunderstood what you were saying.

I assumed that you meant letting `//test/config_test:example_configs_test` pass without reading certificates from disk, but after some digging, I've noticed that `//test/tools/config_load_check:config_load_check_tool` uses the same code, so you probably meant that you're using this tool to verify production configs and want to test them without access to the real certificates? Is that correct?

Reply:
We actually use both a private version of the config test as well as the config load check tool (in different places). We need both to run in a mode that doesn't require the actual cert files to exist (because yes we test prod configs where the certs don't live). IMO it's fine to just have them both work the same way for now (don't load) which is not a regression from what happened before, so any kind of mock is fine. The optimal solution would be to not use a full mock TLS context loader so we can check other config options, but to only mock the actual cert loading part. I have no idea if that is easy or not.

Reply:
Instead of having some code to avoid validating certain parts of the config (which will cause unexpected/unintuitive results), would it be possible to generate valid-ish certs (self-signed or something) in your config validation?  Otherwise we're hard-coding lyft-specific behavior into the config test, which may or may not be what other users need.  I think at a minimum it should by default validate the cert (assume it is present), and there can be a flag or something to opt-in to the behavior of not validating this part of the config.

Reply:
We can open an issue to do what you describe, but if we want to merge this sooner we need to not break existing behavior. IMO having a flag on whether to check the cert or not makes a lot of sense.

Reply:
I agree with @ggreenway, it's one thing to mock things in tests, but faking certificates when testing production config sounds like a really bad idea, and if you're testing those on a machine without access to real certificates, then you should provide self-signed certs. Otherwise, you might refer to certificates that don't exist (e.g. typo).

Furthermore, we'd need to add same hacks to `--mode validate`, which sounds even worse.

Yes, it's change of current behavior, but I don't see how this justifies adding error-prone hacks to the code.

Reply:
I hear what you are saying, but, that's the way it works today, and this PR will break us. If we want to hold this PR until we fix the way we test things, we can do that, but it will take 1-2 weeks since we will need to schedule the work.

Reply:
I'd prefer to wait, then.

Reply:
Alright I will make an internal ticket and get someone to work on cutting us over to fake certs for testing next week.